validator/src/main.rs

```diff
@@ -1374,7 +1375,11 @@ pub fn main() {
         },
         staked_nodes_overrides: staked_nodes_overrides.clone(),
         replay_slots_concurrently: matches.is_present("replay_slots_concurrently"),
-        boot_from_local_state: matches.is_present("boot_from_local_state"),
+        boot_from_local_state: value_t_or_exit!(
+            matches,
+            "use_snapshot_archives_at_startup",
+            UseSnapshotArchivesAtStartup
+        ) == UseSnapshotArchivesAtStartup::Never,
```

Comment:
Is the `==` thing a custom operator on the return value of `value_t_or_exit!()`? Haven't run  into that construct before

Reply:
`value_t_or_exit` just returns the enum value. it's just a one-liner to compare using the Eq of the enum-type. Equivalent to:
```rust
let use_snapshot_archives_at_startup = value_t_or_exit!(matches, "use_snapshot_archives_at_startup", UseSnapshotArchivesAtStartup);
let boot_from_local_state = use_snapshot_archives_at_startup == UseSnapshotArchivesAtStartup::Never;
```

Reply:
lol oh sorry, of course 🫠
I was reading it weird, maybe I need to hit my head again